### DISCUSSION TITLE
[Question] How to have the Rigid Body Constraint of an Animation Nodes's Empty, point to other instances instead of the original objects?

### DISCUSSION THREAD

**Discussion by dentex** [JacquesLucke/animation_nodes#1796] [Q&A] [answered]
I know the question might be a bit convoluted, so I'll try to explain here what I'm doing.

I'm using Blender 2.93.5 and Animation Nodes 2.3.0.

I want to realize, in 3D, an equalizer-style audio visualization, where each frequency bar tip has a smaller item bouncing on top; something like this:

[![eq-example][1]][1]

I have the working version without bouncing tips...

[![v1][2]][2]

...and I have a working example for the moving tip, constrained using an `Empty` with `Rigid Body Constraint` physics enabled on "Generic", with only the Z Axis free to move:

[![original objs][3]][3]

Now the problem is that when instantiating all the eq bars, every `Empty` instance continue to point to the original "center" and "tip" objects, resulting in a mess bouncing all around:

[![mess][4]][4]

I can manually make every `Empty` instance point to the correct instantiated obj, instead of the original one...

[![man-edit][5]][5]

...here i.e. the `Empty.011` is highlighted and I edited `Cube.Tip` to be `Cube.Tip.011`.

Unfortunately this solution is not *stable*, because as soon as I move something and the animation nodes editor updates, the manual edit is being overwritten (i.e. if I increase the number of eq bars).

Is there a solution to this?

Thanks.

[<img src="https://blend-exchange.com/embedImage.png?bid=9ZGBmlnv" />](https://blend-exchange.com/b/9ZGBmlnv/)

  [1]: https://i.stack.imgur.com/ENJ1n.png
  [2]: https://i.stack.imgur.com/fLd21.png
  [3]: https://i.stack.imgur.com/YyCwe.png
  [4]: https://i.stack.imgur.com/ZGyPN.png
  [5]: https://i.stack.imgur.com/o7DDo.png

**Comment by OmarEmaraDev** [comment]
You can just set those objects in the node tree using the *Object Attribute Output* node using the following attributes:

![20211017-165222](https://user-images.githubusercontent.com/25300994/137632824-4b65917b-dced-400c-94b0-7e53e7ee554e.png)

However, I think what you really need is something like this:

![20211017-170040](https://user-images.githubusercontent.com/25300994/137633082-7635cc5c-7070-48bd-94ff-dd9e1712ff46.png)

We create two identical sound spectrum nodes, except one has a higher release time, which corresponds to a lagging effect. The one with the higher release controls the location of the bouncy object.

**Comment by dentex** [comment, marked as answer]
Solution one:
I used the `Object Matrix Output` to make it work, like this (checking "multiple values"):
![immagine](https://user-images.githubusercontent.com/2931982/137638438-231431bd-1446-4a6f-9d3c-573987168fcd.png)
and all objects are instantiated correctly... but then the "tips" never falls and/or Blender crashes every time.

Solution two:
On the other hand, this is smarter. Drop the rigid body physics...
The effect it's not exactly the same as with the tip "falling" due to gravity... but whatever.

Many thanks for the great answers!   :)
